src/sentry/integrations/notifydisable.py

```diff
@@ -0,0 +1,57 @@
+from sentry import integrations
+from sentry.models import Organization
+from sentry.services.hybrid_cloud.integration import RpcIntegration
+from sentry.utils.email import MessageBuilder
+
+provider_types = {
+    "integration": "integrations",
+    "plugin": "plugins",
+    "sentry-app": "sentry-apps",
+}
+
+
+def get_url(organization: Organization, provider_type: str, provider: str) -> str:
+    type_name = provider_types.get(provider_type, "")
+    return str(
+        organization.absolute_url(
+            f"/settings/{type_name}/{provider}/",
+        )
+    )
+
+
+def get_provider_type(redis_key: str) -> str:
+    for provider in provider_types:
+        if provider in redis_key:
+            return provider
+
+    return ""
+
+
+def get_subject(integration_name: str) -> str:
+    return f"Action required: re-authenticate or fix your {integration_name} integration"
+
+
+def notify_disable(
+    organization: Organization, integration: RpcIntegration, redis_key: str, project: None
+):
+
+    provider = integrations.get(integration.provider)
+    integration_name = provider.name
+    integration_link = get_url(
+        organization,
+        get_provider_type(redis_key),
```

Comment:
Maybe what you can do is call `get_provider_type` outside of this and if it's an empty string, set the `integration_link` to the integration directory. Like ...

```
provider = integrations.get(integration.provider)
integration_name = provider.name
provider_type = get_provider_type(redis_key)
integration_link = get_url(....)

# and then in get_url do:
if not provider_type:
   return str(organization.absolute_url(f"/settings/{type_name}//")
return #what you're already returning
```

and make sure that `get_provider_type` returns an empty string if it's not found.